cloud_storage_handler/clients/minio_client.py

```diff
@@ -22,21 +24,44 @@ def __init__(self, config_file=None):
                                uses the default path.
         """
         config_file = config_file or get_config_path()
-        self.config = Foca(config_file=config_file).conf
+        self.config: CustomConfig = Foca(config_file=config_file).conf
 
     def get_minio_config(self) -> MinioConfig:
         """Retrieve the MinIO configuration.
 
         Returns:
             A dictionary containing the MinIO configuration.
         """
-        minio_config = self.config.custom.get("minio")
+        minio_from_foca_config = self.config.minio
         return MinioConfig(
-            hostname=minio_config["hostname"],
-            port=minio_config["port"],
-            access_key=minio_config["access_key"],
-            secret_key=minio_config["secret_key"],
-            bucket_name=minio_config["bucket_name"],
+            hostname=getattr(
+                minio_from_foca_config,
+                "hostname",
+                chs_constants.default_minio_config.hostname,
+            ),
+            port=getattr(
+                minio_from_foca_config, "port", chs_constants.default_minio_config.port
+            ),
+            access_key=getattr(
+                minio_from_foca_config,
+                "access_key",
+                chs_constants.default_minio_config.access_key,
+            ),
+            secret_key=getattr(
+                minio_from_foca_config,
+                "secret_key",
+                chs_constants.default_minio_config.secret_key,
+            ),
+            bucket_name=getattr(
+                minio_from_foca_config,
+                "bucket_name",
+                chs_constants.default_minio_config.bucket_name,
+            ),
+            secure=getattr(
+                minio_from_foca_config,
+                "secure",
+                chs_constants.default_minio_config.secure,
+            ),
```

Comment:
This ensures that the config is either coming from FOCA or the default, preferring FOCA.